File: java/socketd/src/main/java/org/noear/socketd/core/entity/FileEntity.java
package org.noear.socketd.core.entity;

import org.noear.socketd.core.Constants;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

/**
 * 文件实体
 *
 * @author noear
 * @since 2.0
 */
public class FileEntity extends StreamEntity {

    public FileEntity(File file) throws IOException {
        super(new FileInputStream(file));
        putMeta(Constants.META_DATA_DISPOSITION_FILENAME, file.getName());
    }
}

File: python/socketd/socketd/core/module/FileEntity.py

from socketd.core.module.Entity import EntityMetas
from socketd.core.module.EntityDefault import EntityDefault


class FileEntity(EntityDefault):

    def __init__(self, byteIO: bytes, filename: str):
        super().__init__()
        self._file = byteIO
        self.set_data(byteIO)
        self.set_meta(EntityMetas.META_DATA_DISPOSITION_FILENAME, filename)
